operator/slot_ticker/slotticker.go

```diff
@@ -6,83 +6,90 @@ import (
 	"github.com/attestantio/go-eth2-client/spec/phase0"
 )
 
-// The TTicker interface defines a type which can expose a
-// receive-only channel firing slot events.
-type TTicker interface {
-	C() <-chan phase0.Slot
-	Done()
+//go:generate mockgen -package=mocks -destination=./mocks/slotticker.go -source=./slotticker.go
+
+type SlotTickerProvider func() SlotTicker
+
+type SlotTicker interface {
+	Next() <-chan time.Time
+	Slot() phase0.Slot
+}
+
+type ConfigProvider interface {
+	SlotDurationSec() time.Duration
+	GetGenesisTime() time.Time
 }
 
-// SlotTicker is a special ticker for the beacon chain block.
-// The channel emits over the slot interval, and ensures that
-// the ticks are in line with the genesis time. This means that
-// the duration between the ticks and the genesis time are always a
-// multiple of the slot duration.
-// In addition, the channel returns the new slot number.
-type SlotTicker struct {
-	c    chan phase0.Slot
-	done chan struct{}
+type SlotTickerConfig struct {
```

Comment:
same comment, this could be just `Config` so end of the day it is used as `slot_ticker.Config`, more compact and more readable.

Reply:
Let's maybe also rename `slot_ticker` as well as underscore it usually not used in package names?